### ISSUE TITLE
Header response validation

### ISSUE THREAD

**Issue by akatsukle** [anttiviljami/openapi-backend#16]
In #7, a mechanism to allow output validation was added using `postResponseHandler`. I like how its working since it let the responsibility to call `validateResponse` or not to the application.

In the OpenAPI specs, its also possible to define headers that we're expecting in the response.
I think it would make sense to add a validateResponseHeader method to the same location, but what are your thought on this?

I could work on a PR for it if you are interested.

**Comment by anttiviljami** [comment]
I like it!

If you decide to work on the PR for this, a couple of suggestions:

- By default, the validation for response headers should probably have `coerceTypes: true` similarly to request headers
- This means you will need separate validation functions for the response body and headers
- To differentiate the error messages for request body and headers, you should probably validate the schema with a root object + named property something like this:
```
const resBodySchema = {
  body: { oneOf: resBodySchemas },
};
const resHeaderSchema {
  header: { oneOf: resHeaderSchemas },
};
```

vs. right now the response body validation has no root object:
```
// compile the validator function and register to responseValidators
const schema = { oneOf: responseSchemas };
```

Great idea! :)

**Comment by anttiviljami** [comment]
Fixed & merged #18
